documentation/modules/oauth/con-oauth-authorization-keycloak-authorization-services.adoc

```diff
@@ -0,0 +1,199 @@
+
+.Authorization scopes
+
+Typically the initial configuration involves uploading the authorization scopes which creates a list of all the possible actions that can be performed on all the Kafka resource types.
+This step is performed only once, before defining any permissions. Alternatively, the authorization scopes can be added manually, but make sure to not introduce typos.
+
+The `authorization scopes` should simply contain all the possible Kafka permissions regardless of the resource type:
+
+* Create
+* Write
+* Read
+* Delete
+* Describe
+* Alter
+* DescribeConfig
+* AlterConfig
+* ClusterAction
+* IdempotentWrite
+
+.Resources
+
+The `resources` use pattern names to be used with pattern matching against the targeted resources when performing the permission check.
+
+The general pattern is as follows: `RESOURCE_TYPE:NAME_PATTERN`
+
+The resource types mirror the Kafka authorization model.
+The pattern allows for the two matching options: exact matching (when the pattern does not end with `\*`), and prefix matching (when the pattern ends with `*`).
+
+A few examples:
+
+    Topic:my-topic
+    Topic:orders-*
+    Group:orders-*
+    Cluster:*
+
+In addition, the general pattern can be prefixed by another one of the format `kafka-cluster:CLUSTER_NAME`, followed by comma, where the cluster name refers to the `metadata.name` in the Kafka CR.
+
+For example:
+
+    kafka-cluster:my-cluster,Topic:*
+    kafka-cluster:*,Group:b_*
+
+When the `kafka-cluster` prefix is not present it is assumed to be `kafka-cluster:*`.
+
+When the resource is defined, a list of possible authorization scopes relevant to the resource should be added to the list of scopes.
+Set whatever actions make sense for the targeted resource type.
+
+While you may add any `authorization scope` to any `resource`, only the scopes supported by the resource type will ever matter.
+
+.Policies
+
+The `policies` are used to target permissions to one or more accounts.
+The targeting can refer to specific user or service accounts, it can refer to the realm roles or client roles, it can refer to user groups, and it can even use a JS rule and match client's IP address for example.
+
+Each policy can be given a name, and can be reused to target multiple permissions to multiple resources.
+
+.Permissions
+
+The `permissions` are the final step where you pull together the `policies`, `resources`, and `authorization scopes` to grant access to one or more users.
+
+Scope permissions should be used to grant fine-grained permissions to users.
+
+Each policy should be descriptively named in order to make it very clear what permissions it grants to which users.
+
+See xref:con-oauth-authorization-keycloak-example_str[the authorization example] to get a hands-on understanding of how to configure the permissions through Keycloak Authorization Services.
+
+
+== Permissions required by operations
+
```

Comment:
We need some context here. What are describing in this section and how it's relevant to the reader.

Reply:
Added the next line.